airflow/models.py

```diff
@@ -3438,7 +3438,8 @@ def refresh_from_db(self, session=None):
 
     @staticmethod
     @provide_session
-    def find(dag_id, run_id=None, state=None, external_trigger=None, session=None):
+    def find(dag_id, run_id=None, state=None, external_trigger=None, session=None,
+             execution_date=None):
```

Comment:
@bolkedebruin this touches your method. I think it's fine (even though landscape complains about >5 args), but wanted to FYI you, since you're messing with this stuff as part of the scheduler roadmap work as well.